### PR TITLE
[Feature] `featmap_vis_demo` supports input image folder and url reference

### PR DESCRIPTION
**Pull request by PeterH0323**
## Motivation

Supports input image folder and url reference

## Modification

`demo/featmap_vis_demo.py`

### CODE REVIEW

File: demo/image_demo.py [hunk 1]
```diff
@@ -55,7 +57,7 @@ def main(args):
 
     is_dir = os.path.isdir(args.img)
     is_url = args.img.startswith(('http:/', 'https:/'))
-    is_file = os.path.splitext(args.img)[-1] in (IMG_EXTENSIONS)
+    is_file = os.path.splitext(args.img)[-1] in IMG_EXTENSIONS
```
Comment (Nioolek) [inline demo/image_demo.py:60]: ->os.path.splitext(args.img)[-1].lower() in IMG_EXTENSIONS

File: demo/image_demo.py [hunk 2]
```diff
@@ -55,7 +57,7 @@ def main(args):
 
     is_dir = os.path.isdir(args.img)
```
Comment (hhaAndroid) [inline demo/image_demo.py:58]: Considering that this function will be used in many places, it is recommended to move it to util and reuse it

File: demo/utils.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# Copyright (c) OpenMMLab. All rights reserved.
```
Comment (hhaAndroid) [inline demo/utils.py:1]: 这个文件不应该放这，而是 mmyolo 内部去

File: mmyolo/utils/misc.py [hunk 4]
```diff
@@ -9,3 +19,103 @@ def switch_to_deploy(model):
...
+        source_file_path_list = [source_root]
+    else:
+        print('Cannot find image file.')
+
+    source_type = dict(is_dir=is_dir, is_url=is_url, is_file=is_file)
+
+    return source_file_path_list, source_type
+
+
+def gen_out_file_path(source_path: str,
```
Comment (hhaAndroid) [inline mmyolo/utils/misc.py:93]: 这个没有啥通用性，没有必要写成函数了

File: mmyolo/utils/misc.py [hunk 5]
```diff
@@ -9,3 +19,103 @@ def switch_to_deploy(model):
...
+    source_file_path_list = []
+    if is_dir:
+        # when input source is dir
+        for file in scandir(source_root, IMG_EXTENSIONS, recursive=True):
+            source_file_path_list.append(os.path.join(source_root, file))
+    elif is_url:
+        # when input source is url
+        filename = os.path.basename(
+            urllib.parse.unquote(source_root).split('?')[0])
+        torch.hub.download_url_to_file(source_root, filename)
```
Comment (hhaAndroid) [inline mmyolo/utils/misc.py:80]: 要去验证下，这种写法图片会下载到哪里，不会是  mmyolo/utils 下吧

Comment (PeterH0323) [reply mmyolo/utils/misc.py]: 下载到 mmyolo 项目根目录中

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**